game-app/game-core/src/main/java/games/strategy/triplea/ui/UnitChooser.java

```diff
@@ -201,6 +231,12 @@ private void createEntries(
       final Collection<Unit> units,
       final UnitSeparator.SeparatorCategories separatorCategories,
       final Collection<Unit> defaultSelections) {
+    spaceRequiredForNonWithdrawableIcon = units
+        .stream()
+        .anyMatch(unit->
+            Properties.getPartialAmphibiousRetreat(unit.getOwner().getData().getProperties())
```

Comment:
It's dangerous to get a 'gameData' from a 'GamePlayer' object. The 'getOwner()' could be null. It's probably better to get `GameData` from `UiContext`.

Reply:
How do I get `GameData` from `UiContext`?

Reply:
Not a good suggestion from me to get GameData from UiContext, that is not trivial. The code is badly structured, the 'unit' should know if partial retreat is possible and should not be coming from a static method that needs 'GameData'.

The better fix is to add a method 'canPartialAmphibiousRetreat' or something like that. The concept of a 'partial' amphib retreat is not clear.

Bottom line, don't worry about this one, but do test thoroughly for neutral units. Ensure that attacking them is fine, that they are fine in battle calc.

Reply:
I'll add a check against getOwner() == null in the next commit

Reply:
That could be debatable. If the 'owner' were null then what would the value be for 'partial amphib retreat'
IMO it's okay to leave, there is an implicit assumption though that 'unit' will never be a neutral (It might be better to even assert that 'getOwner()' != null and if not true then we could at least throw our own exception and print the unit details). Ultimately there is a contract problem here where all units should have an owner but do not (so ultimately that would be the thing to fix, until then we pay the price)